controllers/storagecluster/storageconsumer.go

```diff
@@ -98,26 +116,30 @@ func (s *storageConsumer) ensureCreated(r *StorageClusterReconciler, storageClus
 		return ctrl.Result{}, fmt.Errorf("failed to get available services configured in StorageCluster: %v", err)
 	}
 	consumerConfigMap := &corev1.ConfigMap{}
-	consumerConfigMap.Name = localStorageConsumerConfigMapName
+	consumerConfigMap.Name = storageConsumer.Spec.ResourceNameMappingConfigMap.Name
 	consumerConfigMap.Namespace = storageCluster.Namespace
 	if _, err := controllerutil.CreateOrUpdate(r.ctx, r.Client, consumerConfigMap, func() error {
 		data := util.GetStorageConsumerDefaultResourceNames(
-			defaults.LocalStorageConsumerName,
+			storageConsumer.Name,
 			string(storageConsumer.UID),
 			availableServices,
 		)
 		resourceMap := util.WrapStorageConsumerResourceMap(data)
-		resourceMap.ReplaceRbdRadosNamespaceName(util.ImplicitRbdRadosNamespaceName)
-		resourceMap.ReplaceSubVolumeGroupName(subVolumeGroupName)
-		resourceMap.ReplaceSubVolumeGroupRadosNamespaceName(subVolumeGroupName)
-		resourceMap.ReplaceRbdClientProfileName("openshift-storage")
-		resourceMap.ReplaceCephFsClientProfileName("openshift-storage")
-		resourceMap.ReplaceNfsClientProfileName("openshift-storage")
-		// NB: Do we need to allow user changing/overwriting any values in this configmap?
+		if consumerUID != "" {
```

Comment:
```suggestion
		if consumerUidFromStorageCluster != "" {
```

since consumerUid is generic and uid will only be empty when object doesn't exist which is used in different scenario?

Reply:
Updated it to backwardCompatibleValue as per https://github.com/red-hat-storage/ocs-operator/pull/3157#discussion_r2166569267